testcontainers-dapr/src/main/java/io/dapr/testcontainers/DaprContainer.java

```diff
@@ -303,11 +353,33 @@ protected void configure() {
       this.schedulerContainer.start();
     }
 
+    dependsOn(placementContainer, schedulerContainer);
+  }
+
+  @Override
+  public void start() {
     List<String> cmds = new ArrayList<>();
     cmds.add("./daprd");
+    
+    String finalAppName = (appName != null && !appName.trim().isEmpty())
+        ? appName 
+        : "dapr-app-" + UUID.randomUUID().toString().substring(0, 8);
+    
     cmds.add("--app-id");
-    cmds.add(appName);
+    cmds.add(finalAppName);
+    
     cmds.add("--dapr-listen-addresses=0.0.0.0");
+    
+    if (customHttpPort != null) {
+      cmds.add("--dapr-http-port");
+      cmds.add(Integer.toString(customHttpPort));
+    }
+    
+    if (customGrpcPort != null) {
```

Comment:
@cicoyle I really think we shouldn't do this. As we are relying on Testcontainers to map the ports, there is no need to change the default ones.